2-TokenCache/Console-TokenCache/Program.cs

```diff
@@ -0,0 +1,218 @@
+﻿using Microsoft.Extensions.Configuration;
+using Microsoft.Graph;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+using System;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+
+namespace Console_TokenCache
+{
+    class Program
+    {
+        private static PublicClientApplicationOptions appConfiguration = null;
+        private static IConfiguration configuration;
+        private static string _authority;
+
+        static async Task Main(string[] args)
+        {
+            // Using appsettings.json as our configuration settings
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(System.IO.Directory.GetCurrentDirectory())
+                .AddJsonFile("appsettings.json");
+
+            configuration = builder.Build();
+
+            appConfiguration = configuration
+                .Get<PublicClientApplicationOptions>();
+
+            _authority = string.Concat(appConfiguration.Instance, appConfiguration.TenantId);
+
+            // Building a public client application
+            var app = PublicClientApplicationBuilder.Create(appConfiguration.ClientId)
+                                                    .WithAuthority(_authority)
+                                                    .WithRedirectUri(appConfiguration.RedirectUri)
+                                                    .Build();
+
+            // Building StorageCreationProperties
+            var storageProperties =
+                 new StorageCreationPropertiesBuilder(CacheSettings.CacheFileName, CacheSettings.CacheDir, appConfiguration.ClientId)
+                 .WithLinuxKeyring(
+                     CacheSettings.LinuxKeyRingSchema,
+                     CacheSettings.LinuxKeyRingCollection,
+                     CacheSettings.LinuxKeyRingLabel,
+                     CacheSettings.LinuxKeyRingAttr1,
+                     CacheSettings.LinuxKeyRingAttr2)
+                 .WithMacKeyChain(
+                     CacheSettings.KeyChainServiceName,
+                     CacheSettings.KeyChainAccountName)
+                 .Build();
+
+            // This hooks up the cross-platform cache into MSAL
+            var cacheHelper = await MsalCacheHelper.CreateAsync(storageProperties);
+            cacheHelper.RegisterCache(app.UserTokenCache);
+
+            // Subscribing to the CacheChanged event
```

Comment:
You should really mention that this event is computationally and IO expensive and should be used with care. Its main use is for UI apps to be able to update the status of the widget that displays accounts.

In most cases you don't need this event - it's really only usefuly for UI apps sharing the same token cache.

Reply:
yes, I was wondering about the usefulness of this event in the case of this tutorial. Maybe we should just remove it alltogether?

Reply:
Removed